core/api/src/app/accounts/get-default-wallet-for-username-phone.ts

```diff
@@ -0,0 +1,28 @@
+import { getWalletFromAccount } from "./get-wallet-from-account"
+
+import { checkedToUsername } from "@/domain/accounts"
+import { checkedToPhoneNumber } from "@/domain/users"
+import { AccountsRepository, UsersRepository } from "@/services/mongoose"
+
+export const getDefaultWalletByUsernameOrPhone = async (
+  value: Username | PhoneNumber,
+  walletCurrency?: WalletCurrency,
+): Promise<Wallet | RepositoryError> => {
+  const checkedUsername = checkedToUsername(value)
+  if (!(checkedUsername instanceof Error)) {
+    const account = await AccountsRepository().findByUsername(checkedUsername)
+    if (account instanceof Error) return account
+    return getWalletFromAccount(account, walletCurrency, value)
+  }
+
+  const checkedPhoneNumber = checkedToPhoneNumber(value)
+  if (checkedPhoneNumber instanceof Error) return checkedPhoneNumber
```

Comment:
although this is the usual pattern, in this case please return `CouldNotFindWalletFromUsernameAndCurrencyError(value)`
```suggestion
  if (checkedPhoneNumber instanceof Error) return new CouldNotFindWalletFromUsernameAndCurrencyError(value)
```